src/adapter/src/coord.rs

```diff
@@ -1340,6 +1345,8 @@ pub struct Coordinator {
 
     /// A map from active subscribes to the subscribe description.
     active_subscribes: BTreeMap<GlobalId, ActiveSubscribe>,
+    /// A map from active copy tos to the copy to description.
+    active_copy_tos: BTreeMap<GlobalId, ActiveCopyTo>,
```

Comment:
High level comment, don't let this derail you. It seems like at least from the Adapter's perspective there are a lot of similarities between active subscribes and active copy_tos, could it simply some handling if we unify them? Maybe something like:

```
// Active dataflows that are sinking (outputting) data and are short lived, e.g.
// their lifetime is tied to a session or they have a finite amount of data.
enum ActiveEphemeralSink {
  Subscribe(...),
  CopyTo(...),
}

// ...

active_ephemeral_sinks: BTreeMap<GlobalId, ActiveEphemeralSink>,
```

Reply:
That would be nice! I will give it a stab

Reply:
One take from reading the existing code: there's enough changing here that I would recommend doing this refactor in a followup PR to reduce the size of this PR and mental complexity of us reading what's going on. Adding this refactor will make it harder for you and the reviewers to figure out what's going on, because we'd now have to consider SUBSCRIBE.

Reply:
I just got done with the refactor 🙈. I would say the diffs are not too bad, rather contrasts with subscribe nicely. Since they are in separate commits, I can also spin it off as a different PR if needed.

Reply:
The wrapper `ActiveComputeSink` is refactored and can be reviewed in a separate commit https://github.com/MaterializeInc/materialize/pull/25042/commits/d501f03cd870ff26da70e7191da2591dcad0b3fc